pkg/scheduler/core/spreadconstraint/select_clusters_by_labels.go

```diff
@@ -0,0 +1,63 @@
+package spreadconstraint
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+)
+
+// selectBestClustersByClusterLabels support for SpreadByLabel constraints.
+func selectBestClustersByClusterLabels(spreadConstraint []policyv1alpha1.SpreadConstraint, groupClustersInfo *GroupClustersInfo,
+	needReplicas int32) ([]*clusterv1alpha1.Cluster, error) {
+	var clusterInfos = groupClustersInfo.Clusters
+	for i := range spreadConstraint {
+		if spreadConstraint[i].SpreadByLabel == "" {
+			continue
+		}
+		// firstly, match cluster labels
+		requirement, err := labels.NewRequirement(spreadConstraint[i].SpreadByLabel, selection.Exists, []string{})
+		if err != nil {
+			return nil, err
+		}
+		selector := labels.NewSelector().Add(*requirement)
+		var selectedClusters []ClusterDetailInfo
+		for j := range clusterInfos {
+			if selector.Matches(labels.Set(clusterInfos[j].Cluster.GetLabels())) {
+				selectedClusters = append(selectedClusters, clusterInfos[j])
+			}
```

Comment:
I have a question if two clusters have the same label(same key and value), should they be considered one group or two groups?

Reply:
one group.